drv/auxflash-api/src/lib.rs

```diff
@@ -171,6 +171,55 @@ where
     }
 }
 
+/// Extension functions on the autogenerated `AuxFlash` type
+impl AuxFlash {
+    /// Reads a compressed blob, streaming it to a callback function
+    pub fn get_compressed_blob_streaming<
+        F: Fn(&[u8]) -> Result<(), E>,
+        E: From<AuxFlashError>,
+    >(
+        &self,
+        tag: [u8; 4],
+        f: F,
+    ) -> Result<[u8; 32], E> {
+        let blob = self.get_blob_by_tag(tag)?;
+        let mut scratch_buf = [0u8; 128];
+        let mut pos = blob.start;
+        let mut sha = Sha3_256::new();
+        let mut decompressor = gnarle::Decompressor::default();
+
+        while pos < blob.end {
+            let amount = (blob.end - pos).min(scratch_buf.len() as u32);
+            let chunk = &mut scratch_buf[0..(amount as usize)];
+            self.read_slot_with_offset(blob.slot, pos, chunk)?;
+            sha.update(&chunk);
+            pos += amount;
+
+            // Reborrow as an immutable chunk, then decompress
+            let mut chunk = &scratch_buf[0..(amount as usize)];
+            let mut decompress_buffer = [0; 512];
+
+            while !chunk.is_empty() {
+                let decompressed_chunk = gnarle::decompress(
+                    &mut decompressor,
+                    &mut chunk,
+                    &mut decompress_buffer,
+                );
+
+                // The compressor may have encountered a partial run at the
```

Comment:
nit: shouldn't this say
```suggestion
                // The decompressor may have encountered a partial run at the
```

Reply:
thanks, fixed